dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/SampleToBatchPadding.scala

```diff
@@ -0,0 +1,139 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.dataset.text
+
+import java.util
+
+import com.intel.analytics.bigdl.dataset._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Storage, Tensor}
+
+import scala.collection.Iterator
+import scala.reflect.ClassTag
+
+object SampleToBatchPadding {
+  def apply[T: ClassTag]
+  (totalBatch : Int,
+   padding : Boolean = false,
+   paddingValue : Option[T] = None,
+   paddingLength: Option[Int] = None)
+  (implicit ev: TensorNumeric[T]): SampleToBatchPadding[T]
+  = new SampleToBatchPadding[T](totalBatch, padding, paddingValue, paddingLength)
+}
+
+/**
+ * convert sample to MiniBatch, and you can choose to do padding or not.
+ * Just padding for feature, not for label.
+ * @param totalBatch batchsize
+ * @param padding whether padding to the same length in a batch
+ * @param paddingValue padding value, default 0
+ * @param paddingLength padding length, default the max first dimension length of feature in a batch
+ */
+class SampleToBatchPadding[T: ClassTag]
+(totalBatch : Int,
+ padding : Boolean,
+ paddingValue : Option[T] = None,
+ paddingLength: Option[Int] = None)
+(implicit ev: TensorNumeric[T])
+  extends Transformer[Sample[T], MiniBatch[T]] {
+
+  override def apply(prev: Iterator[Sample[T]]): Iterator[MiniBatch[T]] = {
+    new Iterator[MiniBatch[T]] {
+      private var featureTensor: Tensor[T] = Tensor[T]()
+      private var labelTensor: Tensor[T] = Tensor[T]()
+      private var featureData: Array[T] = null
+      private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
+      private val batchSize = Utils.getBatchSize(totalBatch)
+      private var featureSize: Array[Int] = null
+      private var labelSize: Array[Int] = null
+      private var oneFeatureLength: Int = 0
+      private var oneLabelLength: Int = 0
+      override def hasNext: Boolean = prev.hasNext
+
+      override def next(): MiniBatch[T] = {
+        if (prev.hasNext) {
+          var i = 0
+          var maxDataSize: Array[Int] = null
+          var maxElement = 0
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
+          while (i < batchSize && prev.hasNext) {
+            val sample = prev.next()
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
```

Comment:
What if the next batchSize is larger, then the sampleData will throw indexOutOfBoundary error?